packages/hackmode/lib/index.js

```diff
@@ -444,26 +737,89 @@ class N3hHackMode extends AsyncClass {
     return this._memory[dnaAddress]
   }
 
+  _checkRequest (requestId) {
+    if (!(requestId in this._requestBook)) {
+      return ''
+    }
+    let bucketId = this._requestBook[requestId]
+    delete this._requestBook[requestId]
+    return bucketId
+  }
+
   _track (dnaAddress, agentId) {
     const ref = this._getMemRef(dnaAddress)
+    // store agent (this will map agentId to transportId)
     ref.mem.insert({
       type: 'agent',
       dnaAddress: dnaAddress,
       agentId: agentId,
       address: agentId,
       transportId: this._p2p.getId()
     })
+
+    // send get lists
+    let requestId = this._createRequest(dnaAddress, agentId)
+    this._ipc.send('json', {
+      method: 'handleGetPublishingEntryList',
+      dnaAddress,
+      _id: requestId
+    })
+    requestId = this._createRequest(dnaAddress, agentId)
+    this._ipc.send('json', {
+      method: 'handleGetHoldingEntryList',
+      dnaAddress,
+      _id: requestId
+    })
+    requestId = this._createRequest(dnaAddress, agentId)
+    this._ipc.send('json', {
+      method: 'handleGetPublishingMetaList',
+      dnaAddress,
+      _id: requestId
+    })
+    requestId = this._createRequest(dnaAddress, agentId)
+    this._ipc.send('json', {
+      method: 'handleGetHoldingMetaList',
+      dnaAddress,
+      _id: requestId
+    })
   }
 
-  _fullGossipHashHash () {
-    const out = []
-    for (let dnaAddress in this._memory) {
-      out.push({
-        dnaAddress,
-        gossipHashHash: this._memory[dnaAddress].mem.getGossipHashHash()
-      })
+  _CatDnaAgent (DnaHash, AgentId) {
+    return '' + DnaHash + '::' + AgentId
+  }
+
+  // Make a metaId out of an entryAddress and an attribute
+  _CatEntryAttribute (entryAddress, attribute) {
+    return '' + entryAddress + '||' + attribute
+  }
+
+  _generateRequestId () {
+    this._requestCount += 1
+    return 'req_' + this._requestCount
+  }
+
+  // create and return a new request_id
+  _createRequest (dnaAddress, agentId) {
+    let bucketId = this._CatDnaAgent(dnaAddress, agentId)
+    return this._createRequestWithBucket(bucketId)
+  }
+
+  _createRequestWithBucket (bucketId) {
+    let reqId = this._generateRequestId()
+    this._requestBook[reqId] = bucketId
+    return reqId
+  }
+
+  _bookkeepAddress (book, dnaAddress, address) {
+    if (!(dnaAddress in book)) {
+      let array = []
+      array.push(address)
+      book[dnaAddress] = array
+      return
     }
-    return out
+    let array = book[dnaAddress]
+    array.push(address)
+    book[dnaAddress] = array
```

Comment:
to avoid some duplication here, i like to check for the array existence first, then always append, like:

```javascript
if (!(dnaAddress in book)) {
  book[dnaAddress] = []
}
book[dnaAddress].push(address)
```

Reply:
done